### PR TITLE
Fix current CI flakiness with DiagnosticLspSuite and LensesLspSuite

### PR DESCRIPTION
**Pull request by tgodzik**

### CODE REVIEW

File: tests/unit/src/test/scala/tests/CodeLensesLspSuite.scala [hunk 1]
```diff
@@ -35,23 +35,24 @@ object CodeLensesLspSuite extends BaseLspSuite("codeLenses") {
...
-
-  check("test-suite-object", library = "com.lihaoyi::utest:0.7.1")(
-    """|package foo.bar
-       |<<test>>
-       |object Foo extends utest.TestSuite {
-       |  val tests = utest.Tests {}
-       |}
-       |""".stripMargin
-  )
+  // broken when being run on a fresh Bloop instance
```
Comment (tgodzik) [inline tests/unit/src/test/scala/tests/CodeLensesLspSuite.scala]: Working on fixing that separately

**Comment by tgodzik** [comment]
Seems we have another flakines:
```
2019-10-17T17:34:04.9163732Z INFO  compiling b (1 scala source)
2019-10-17T17:34:05.9380001Z     """|/a/src/main/scala/a/Main.scala
2019-10-17T17:34:05.9382686Z        |package a
2019-10-17T17:34:05.9383247Z        |import java.util.concurrent.Future/*Future.java*/ // unused
2019-10-17T17:34:05.9383671Z        |import scala.util.Failure/*Try.scala*/ // unused
2019-10-17T17:34:05.9384022Z        |object Main/*MainSuite.scala:6*/ extends App/*App.scala*/ {
2019-10-17T17:34:05.9384424Z        |  val message/*L6*/ = Message/*Message.java:1*/.message/*Message.java:2*/
2019-10-17T17:34:05.9384827Z        |  new java.io.PrintStream/*PrintStream.java*/(new java.io.ByteArrayOutputStream/*ByteArrayOutputStream.java*/())
2019-10-17T17:34:05.9385252Z        |  println/*Predef.scala*/(message/*L4*/)
2019-10-17T17:34:05.9385622Z        |}
2019-10-17T17:34:05.9386050Z        |/b/src/main/scala/a/MainSuite.scala
2019-10-17T17:34:05.9386419Z        |package a
2019-10-17T17:34:05.9386770Z        |import java.util.concurrent.Future/*Future.java*/ // unused
2019-10-17T17:34:05.9387136Z        |import scala.util.Failure/*Try.scala*/ // unused
2019-10-17T17:34:05.9387502Z        |import org.scalatest.FunSuite/*FunSuite.scala*/
2019-10-17T17:34:05.9391274Z        |object MainSuite/*L4*/ extends FunSuite/*FunSuite.scala*/ {
2019-10-17T17:34:05.9392342Z        |  test/*FunSuiteLike.scala*/("a") {
2019-10-17T17:34:05.9392769Z        |    val condition/*L7*/ = Main/*Main.scala:3*/.message/*Main.scala:4*/.contains/*String.java*/("Hello")
2019-10-17T17:34:05.9393173Z        |    assert/*Assertions.scala*/(condition/*L6*/)
2019-10-17T17:34:05.9393532Z        |  }
2019-10-17T17:34:05.9393865Z        |}
2019-10-17T17:34:05.9464436Z [31mX[39m tests.DefinitionLspSuite.definition [2m10609ms[0m 
2019-10-17T17:34:05.9465594Z   [91m[4mtests.TestFailedException[39m[24m: failed assertion at /home/runner/work/metals/metals/tests/unit/src/test/scala/tests/DefinitionLspSuite.scala:71
2019-10-17T17:34:05.9466195Z failed assertion at /home/runner/work/metals/metals/tests/mtest/src/main/scala/tests/DiffAssertions.scala:24
2019-10-17T17:34:05.9466614Z ===========
2019-10-17T17:34:05.9466973Z => Obtained
2019-10-17T17:34:05.9467317Z ===========
2019-10-17T17:34:05.9467661Z     """|/a/src/main/scala/a/Main.scala
2019-10-17T17:34:05.9468025Z        |package a
2019-10-17T17:34:05.9468386Z        |import java.util.concurrent.Future/*Future.java*/ // unused
2019-10-17T17:34:05.9469321Z        |import scala.util.Failure/*Try.scala*/ // unused
2019-10-17T17:34:05.9469990Z        |object Main/*MainSuite.scala:6*/ extends App/*App.scala*/ {
2019-10-17T17:34:05.9470387Z        |  val message/*L6*/ = Message/*Message.java:1*/.message/*Message.java:2*/
2019-10-17T17:34:05.9471186Z        |  new java.io.PrintStream/*PrintStream.java*/(new java.io.ByteArrayOutputStream/*ByteArrayOutputStream.java*/())
2019-10-17T17:34:05.9471600Z        |  println/*Predef.scala*/(message/*L4*/)
2019-10-17T17:34:05.9471968Z        |}
2019-10-17T17:34:05.9472317Z        |/b/src/main/scala/a/MainSuite.scala
2019-10-17T17:34:05.9472668Z        |package a
2019-10-17T17:34:05.9473022Z        |import java.util.concurrent.Future/*Future.java*/ // unused
2019-10-17T17:34:05.9473399Z        |import scala.util.Failure/*Try.scala*/ // unused
2019-10-17T17:34:05.9473787Z        |import org.scalatest.FunSuite/*FunSuite.scala*/
2019-10-17T17:34:05.9474155Z        |object MainSuite/*L4*/ extends FunSuite/*FunSuite.scala*/ {
2019-10-17T17:34:05.9474514Z        |  test/*FunSuiteLike.scala*/("a") {
2019-10-17T17:34:05.9474915Z        |    val condition/*L7*/ = Main/*Main.scala:3*/.message/*Main.scala:4*/.contains/*String.java*/("Hello")
2019-10-17T17:34:05.9475318Z        |    assert/*Assertions.scala*/(condition/*L6*/)
2019-10-17T17:34:05.9485476Z        |  }
2019-10-17T17:34:05.9485980Z        |}
2019-10-17T17:34:05.9486268Z 
2019-10-17T17:34:05.9486574Z 
```

**Comment by tgodzik** [comment]
Last one looks like the semanticDB document was not created and there are no references - pretty sure it's a race.

**Comment by olafurpg** [comment]
Indexing "find references" from `*.semanticdb` files relies on file watching which is inherently unreliable for stable tests. I'm fine with disabling unstable tests for now since having a green CI is more important IMO.

**Comment by tgodzik** [comment]
> Indexing "find references" from `*.semanticdb` files relies on file watching which is inherently unreliable for stable tests. I'm fine with disabling unstable tests for now since having a green CI is more important IMO.

Does it make sense to search only for definitions in that suite instead of just ignoring it altogether?

The other solution I see is registering for events inside of TestServer and waiting for those. But everything like that heavily relies on changes to metals server itself.

File: tests/unit/src/test/scala/tests/DefinitionLspSuite.scala [hunk 2]
```diff
@@ -74,8 +74,8 @@ object DefinitionLspSuite extends BaseLspSuite("definition") {
            |package a
            |import java.util.concurrent.Future/*Future.java*/ // unused
            |import scala.util.Failure/*Try.scala*/ // unused
-           |object Main/*MainSuite.scala:6*/ extends App/*App.scala*/ {
```
Comment (olafurpg) [inline tests/unit/src/test/scala/tests/DefinitionLspSuite.scala]: I hadn't realized we were calling "find references" in this test suite. That explains why the test was flaky, "find references" triggers a background cascade compilation. Generally, any discarded `Future[T]` is a likely source of flaky test failures. It's a good change to only test for definitions in `DefinitionLspSuite` 👍

**Pull request merged**